echo_server.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python
+
+"""
+Echo Server
+"""
+
+import socket
+import email.utils
+
+
+BUFFERSIZE = 32
+
+
+def echo_server():
+    server_socket = socket.socket(
+        socket.AF_INET,
+        socket.SOCK_STREAM,
+        socket.IPPROTO_IP)
+    server_socket.bind(('127.0.0.1', 50000))
+    server_socket.listen(1)
+
+    try:
+        while True:
+            request = ''
+            done = False
+            conn, addr = server_socket.accept()
+            while not done:
+                msg_part = conn.recv(BUFFERSIZE)
+                if len(msg_part) < BUFFERSIZE:
+                    done = True
+                request += msg_part
+            try:
+                parsed_request = parse_request(request)
+                response = response_ok()
+            except HTTPError400:
+                response = response_error(400, 'HTTP Request malformed.')
+            except HTTPError405:
+                response = response_error(405, 'Not a GET request.')
+            except HTTPError505:
+                response = response_error(505, 'Not HTTP/1.1 protocal.')
```

Comment:
You've absolutely got the right idea here, but you are being more lax with your error codes than you should.  There are [very specific human readable descriptions](http://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html) for each numeric code, and you should always use the right one.  Please fix the descriptions here to be correct.